.github/workflows/quick.yaml

```diff
@@ -16,6 +16,11 @@ on:
     # test PRs targeting this branch code
     branches: [ "master" ]
 
+concurrency:
+  # cancel old runs in case of push to same pr or branch
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
Please do not cancel parent master commit tests when master is advanced (to a child commit) during those parent commit tests. We want every master commit to be fully tested[^1].

[^1]: Currently, we are testing every master commit twice: Once during staging (on `auto` branch), and once after `master` branch is _fast-forwarded_ to that tested staged commit). There is probably a way to optimize that second test away without risking `master` testing integrity, but I have not investigated that optimization, and it may become even more difficult if we decide to publish bootstrapped master sources (à la "daily bootstrapped builds"). We should ignore these repeated tests when defining concurrency groups for this PR.

I did not know enough about `concurrency` to offer a reliable suggestion, but I would start with something like this:

```suggestion
  # Cancel ongoing tests in case of push to the same PR or staging branch,
  # but let previous master commit tests complete.
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: ${{ github.ref != 'master' }}
```

The boolean expression for `cancel-in-progress` is a simplified version of an expression in GitHub [docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-only-cancel-in-progress-jobs-on-specific-branches). I hope I got it right.